ls1intum/Artemis
# `Assessment`: Improve tutor training validation logic to not allow negative score without feedback

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

[ ] * - not needed 

### Motivation and Context


We improve our tutor training logic to not allow tutors to create a feedback with a negative score but no content (text).

### Description

Introduced a new `FeedbackCorrectionErrorType`, which is thrown when tutors feedback has negative score but no detail text. As well as added the required translations.

### Steps for Testing

Prerequisites:
- 1 Tutor

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate to Exercise Assessment Dashboard of an Exercise that has Example Submissions
4. Assess Text Example Submission
  4.1. Put a Negative Score with Empty Feedback on a Text Block
  4.2. Submit the Assessment
  4.3. You should See the "Feedback is required for negative score" error message (see screenshot below)
5. Assess Modeling Example Submission 
  5.1. Put a Negative Score with Empty Feedback on an UML element
  5.2. Submit the Assessment
  5.3. You should See the "Feedback is required for negative score" error message

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots


#### Text
<img width="1282" alt="Screen Shot 2021-11-23 at 20 36 15" src="https://user-images.githubusercontent.com/51077603/143094247-59f32061-50ca-4501-ae0d-aec6b7efb9a7.png">

#### Modeling
<img width="474" alt="Screen Shot 2021-11-23 at 20 37 19" src="https://user-images.githubusercontent.com/51077603/143094253-66b212d2-23e3-4222-ba53-f18412b4a12a.png">
